### ISSUE TITLE
Sharing multiple experiments

### ISSUE THREAD

**Issue by dberenbaum** [iterative/dvc#5489]
`dvc exp push` and `dvc exp pull` each take a single experiment. The experiments command is all about running multiple experiments, so most real-world use cases will involve wanting to push or pull multiple (or even all) experiments. For the initial proof of concept, it made sense to focus on one, but there should be ways to share multiple experiments at once. This could include:

- Specifying a list of experiment IDs
- Combining with `exp list` to get a particular subset of experiments to share
- Sharing all experiments in a branch
- Sharing all experiments from a given parent commit
- Sharing all experiments for the last n commits

**Comment by dberenbaum** [comment]
Related (duplicate?): #4649

**Comment by dberenbaum** [comment]
Seems like this could be handled using flags similar to those found in other dvc commands already:

```
  -w, --workspace       Push experiments derived from the current workspace.
  -a, --all-branches    Push experiments derived from the tip of all Git branches.
  -T, --all-tags        Push experiments derived from all Git tags.
  -A, --all-commits     Push experiments derived from all Git commits.
 ```

The same options could probably apply `dvc exp list`.

**Comment by dberenbaum** [comment]
@pmrowla Are there any issues or complications with pushing multiple experiments that are different from pushing a single experiment?

**Comment by pmrowla** [comment]
@dberenbaum the complication is regarding how to handle conflicts. I suppose if `--force` is given we can just assume the user wants to force push everything, but if `--force` is not used, do we abort as soon as there is any conflict, or do we behave like `dvc push` and continue trying to push everything else?

**Comment by dberenbaum** [comment]
A user requested this in https://discord.com/channels/485586884165107732/485596304961962003/895329378110677003.

**Comment by dberenbaum** [comment]
Moving the priority of this up since we are promoting the collaborative aspect of experiment versioning.

**Comment by pmrowla** [comment]
See https://github.com/iterative/dvc/pull/7131#pullrequestreview-829717037 regarding potentially splitting this issue into separate tasks/steps

**Comment by karajan1001** [comment]
According to https://github.com/iterative/dvc/pull/7131#pullrequestreview-829717037
Split this into four steps.
[ ]  add --rev and updated -n=-1 behavior to exp show
[ ] add the new flags (including -A/--all-commits, -a/--all-branches, --all-tags) to exp list
[ ] make all the flags above to support `pull/push`
[ ] make all the flags above to support `remove`

**Comment by dberenbaum** [comment]
Sorry, @karajan1001. I should have clarified requirements here.

The discussion in #7131 seems similar to the `exp remove` discussion in #6394. Can we take the same approach here, where we only add `--all` for now?

**Comment by karajan1001** [comment]
> Sorry, @karajan1001. I should have clarified requirements here.
> 
> The discussion in #7131 seems similar to the exp remove discussion in #6394. Can we take the same approach here, where we only add --all for now?

of course, it was scheduled in https://github.com/iterative/dvc/issues/5489#issuecomment-992234534.
My original PR had already unified `pull`, `push` and `ls` now we need to extend this to `remove` and unify `show`. And how about `gc` it shares the same arguments but the meaning of them is quite different ( means leave the select ones unchanged while all of the other commands means only act on those selected ones)

**Comment by karajan1001** [comment]
@dberenbaum is there still anything remaining?

**Comment by dberenbaum** [comment]
No, closing this issue, thanks for all your hard work @karajan1001!
